Editor/Utility/CinemachineEditorAnalytics.cs

```diff
@@ -0,0 +1,254 @@
+using System.Reflection;
+using UnityEditor;
+using UnityEngine;
+
+namespace Cinemachine.Editor
+{
+    [InitializeOnLoad]
+    static class CinemachineEditorAnalytics
+    {
+        const int k_MaxEventsPerHour = 1000;
+        const int k_MaxNumberOfElements = 1000;
+        const string k_VendorKey = "unity.cinemachine";
+
+        static Assembly s_CinemachineAssembly;
+
+        // register an event handler when the class is initialized
+        static CinemachineEditorAnalytics()
+        {
+            EditorApplication.playModeStateChanged += SendAnalyticsOnPlayEnter;
+
+            s_CinemachineAssembly = typeof(CinemachineBrain).Assembly;
+        }
+
+        /// <summary>
+        /// Send analytics event when using Create -> Cinemachine menu
+        /// </summary>
+        /// <param name="name">Name of the vcam created</param>
+        public static void SendCreateEvent(string name)
+        {
+            if (!EditorAnalytics.enabled)
+                return;
+
+            var data = new CreateEventData
+            {
+                vcam_created = name,
+            };
+
+            // Register our event
+            EditorAnalytics.RegisterEventWithLimit("cm_create_vcam", k_MaxEventsPerHour, k_MaxNumberOfElements, k_VendorKey);
+
+            // Send the data to the database
+            EditorAnalytics.SendEventWithLimit("cm_create_vcam", data);
+        }
+
+        struct CreateEventData
+        {
+            public string vcam_created; // vcam created from Create -> Cinemachine menu
+        }
+
+        /// <summary>
+        /// Send analytics event when using entering playmode
+        /// </summary>
+        /// <param name="state">State change to detect entering playmode</param>
+        static void SendAnalyticsOnPlayEnter(PlayModeStateChange state)
+        {
+            // Only send analytics if it is enabled
+            if (!EditorAnalytics.enabled)
+                return;
+
+            // Only send data when entering playmode
+            if (state != PlayModeStateChange.EnteredPlayMode)
+                return;
+
+            var startTime = Time.realtimeSinceStartup;
+
+            var cinemachineCore = CinemachineCore.Instance;
+            var vcamCount = cinemachineCore.VirtualCameraCount;
+            var vcamDatas = new List<VcamData>();
+
+            // collect data from all vcams
+            for (int i = 0; i < vcamCount; ++i)
+            {
+                var vcamBase = cinemachineCore.GetVirtualCamera(i);
+                if (vcamBase == null) continue;
+
+                var vcam = vcamBase as CinemachineVirtualCamera;
+                if (vcam != null)
+                {
+                    vcamDatas.Add(ConvertVcamToVcamData(vcam, i.ToString()));
+                }
+                else // Composite vcam (Freelook, Mixing, Statedriven, Clearshot):
+                {
+                    GetExtensions(vcamBase, out List<string> vcamExtensions, out int customExtensionCount);
```

Comment:
What happens when a composite vcam has composite vcams as children?  I think this won't work properly.  Have you tested that case?

Reply:
No, you are right.